src/FlagProvider.tsx

```diff
@@ -23,7 +23,11 @@ const offlineConfig: IConfig = {
 // save startTransition as var to avoid webpack analysis (https://github.com/webpack/webpack/issues/14814)
 const _startTransition = 'startTransition';
 // fallback for React <18 which doesn't support startTransition
-const startTransition = React[_startTransition] || (fn => fn());
+const isReactNative = typeof navigator !== 'undefined' && navigator?.product === 'ReactNative';
```

Comment:
since this detection may not work in every ReactNative context I'm also considering config option disableTransitions

Reply:
`startTransition?: boolean` on provider makes sense. With `useCallback` inside provider.

Reply:
Does it make sense to make an automatic detection that doesn't work all the time? If that's the case I think a config option is a better approach.

What do we use this hook for?

Reply:
The hook was added to solve some hydration error: https://github.com/Unleash/proxy-client-react/pull/125

Reply:
I also prefer config over imperfect feature detection

Reply:
@Tymek your proposal seems fine with optional startTransition. But maybe instead of boolean config we can allow code injection instead?

Reply:
`startTransition={fn => fn()}` would solve the problem in the user space